### PR TITLE
Migrate @superset-ui/color to TypeScript

### PR DESCRIPTION
**Pull request by kristw**
🏆 Enhancements
🏠 Internal

- Migrate `@superset-ui/color` to TypeScript

### CODE REVIEW

File: packages/superset-ui-color/src/CategoricalColorNamespace.ts [hunk 1]
```diff
@@ -1,24 +1,31 @@
...
 import getCategoricalSchemeRegistry from './CategoricalSchemeRegistrySingleton';
 import stringifyAndTrim from './stringifyAndTrim';
 
 export default class CategoricalColorNamespace {
-  constructor(name) {
+  name: string;
+  scales: {
+    [key: string]: CategoricalColorScale;
+  };
+
```
Comment (williaster) [inline packages/superset-ui-color/src/CategoricalColorNamespace.ts]: nit -- could remove space to keep consistent with other component's types

File: packages/superset-ui-color/src/SequentialScheme.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import { scaleLinear } from 'd3-scale';
-import ColorScheme from './ColorScheme';
+import ColorScheme, { ColorSchemeConfig } from './ColorScheme';
 
-function range(count) {
+function range(count: number): number[] {
```
Comment (williaster) [inline packages/superset-ui-color/src/SequentialScheme.ts]: can `number[]` be inferred here? (I know our style guide said rely on inference when possible, this also seems fine to me)

File: packages/superset-ui-core/src/utils/isRequired.ts [hunk 3]
```diff
@@ -1,3 +1,3 @@
-export default function isRequired(field: string) {
+export default function isRequired(field: string): never {
```
Comment (williaster) [inline packages/superset-ui-core/src/utils/isRequired.ts]: hadn't seen `never` before 😮

File: packages/superset-ui-color/package.json [hunk 4]
```diff
@@ -27,6 +27,7 @@
   },
   "dependencies": {
     "@superset-ui/core": "^0.7.0",
+    "@types/d3-scale": "^2.0.2",
```
Comment (williaster) [inline packages/superset-ui-color/package.json]: wow sweet!

File: packages/superset-ui-core/src/utils/isRequired.ts (hunk 3 above)
Comment (kristw) [reply packages/superset-ui-core/src/utils/isRequired.ts]: https://stackoverflow.com/questions/37910669/what-is-the-difference-between-never-and-void-in-typescript

**Pull request merged**